File: gnm-run/mainSlurmPrep.cpp
//
// Created by Max Würfel on 26.06.23.
//

#include "Slurm.h"
# include <string>

// This is required for the key
std::vector<std::string> dSets = {
        "Charlesworth2015",
        "Hennig2011",
        "Demas2006",
        "Maccione2014"
};

// Type of array
std::vector<std::string> meaTypes = {
        "MCS_8x8_200um",
        "MCS_8x8_100um",
        "APS_64x64_42um"
};

int main() {
    // User required parameters
    bool cluster = false;
    int dSet = 0;
    double corrCutoff = 0.2;
    int nSamples = -1;
    int nRuns = 10000;
    double dt = 0.05;

    // Get the dSet Type
    int meaType;
    switch (dSet) {
        case 0: {
            meaType = 0;
            break;
        }
        case 2: {
            meaType = 1;
            break;
        }
        case 3: {
            meaType = 2;
            break;
        }
    };

    // Set the paths
    std::string inDirPath, outDirPath;
    if (cluster) {
        inDirPath = "/store/DAMTPEGLEN/mw894/data/" + dSets[dSet];
        outDirPath = "/store/DAMTPEGLEN/mw894/slurm/" + dSets[dSet];

    } else {
        inDirPath = "/Users/maxwuerfek/code/diss/data/" + dSets[dSet];
        outDirPath = "/Users/maxwuerfek/code/diss/slurm/" + dSets[dSet];
    }

    Slurm::generateInputs(inDirPath,
                          outDirPath,
                          corrCutoff,
                          nSamples,
                          nRuns,
                          dSet,
                          meaType,
                          dt

    );
    return 0;
}

File: gnm/main_slurm_prep.jl
include("slurm.jl")

const PARAMS = Dict(
    "cluster" => false,
    "d_set" => 3,
    "corr_cutoff" => 0.2,
    "n_samples" => -1,
    "n_runs" => 10000,
    "dt" => 0.05
)

const DATA_SETS = Dict(
    1 => "Charlesworth2015",
    2 => "Hennig2011",
    3 => "Demas2006",
    4 => "Maccione2014"
)

const MEA_TYPES = Dict(
    1 => "MCS_8x8_200um",
    2 => "MCS_8x8_100um",
    3 => "APS_64x64_42um"
)

function main()
    # get mea type
    if PARAMS["d_set"] == 1
        mea_type = 1
    elseif PARAMS["d_set"] == 3
        mea_type = 2
    elseif PARAMS["d_set"] == 4
        mea_type = 3
    end

    # set paths 
    if PARAMS["cluster"]
        in_dir = "/store/DAMTPEGLEN/mw894/data/" * DATA_SETS[PARAMS["d_set"]]
        out_dir = "/store/DAMTPEGLEN/mw894/slurm/" * DATA_SETS[PARAMS["d_set"]]
    else
        in_dir = "/Users/maxwuerfek/code/diss/data/" * DATA_SETS[PARAMS["d_set"]]
        out_dir = "/Users/maxwuerfek/code/diss/gnm/slurm/" * DATA_SETS[PARAMS["d_set"]]
    end

    println("Dataset: ", DATA_SETS[PARAMS["d_set"]])
    println("MEA type: ", MEA_TYPES[mea_type])
    println("Runs: ", PARAMS["n_runs"])
    println("Samples: ", PARAMS["n_samples"])

    @time generate_inputs(
        in_dir,
        out_dir,
        PARAMS["n_samples"],
        PARAMS["n_runs"],
        PARAMS["d_set"],
        mea_type,
        PARAMS["dt"],
        PARAMS["corr_cutoff"]
    )
end

main()
